### PR TITLE
Fix build of socket.rs on FreeBSD

### PR DESCRIPTION
**Pull request by yonas**
Use OS X code for FreeBSD as well.

OS: `FreeBSD 14`
Sozu: e451177

### CODE REVIEW

**Comment by FlorentinDUBOIS** [comment]
Hello @yonas, thanks you for this contribution

File: bin/src/main.rs [hunk 1]
```diff
@@ -163,7 +163,7 @@ fn set_workers_affinity(workers: &Vec<WorkerSession>) {
 /// Can bind multiple processes to a CPU core if there are more processes
 /// than CPU cores. Only works on Linux.
 #[cfg(not(target_os = "linux"))]
-fn set_workers_affinity(_: &Vec<Worker>) {}
+fn set_workers_affinity(_: &Vec<cli::SubCmd>) {}
```
Comment (Keksoj) [inline bin/src/main.rs:166]: please import the Worker type instead. Or is there a reason you switched to this type?

File: bin/src/util.rs [hunk 2]
```diff
@@ -118,18 +121,18 @@ pub fn get_config_file_path(args: &cli::Args) -> Result<&str, UtilError> {
 pub unsafe fn get_executable_path() -> Result<String, UtilError> {
     let mut capacity = PATH_MAX as usize;
     let mut path: Vec<u8> = Vec::with_capacity(capacity);
-    path.extend(repeat(0).take(capacity));
+    path.extend(std::iter::repeat(0).take(capacity));
```
Comment (Keksoj) [inline bin/src/util.rs:124]: can you import those at the top of the file instead?

**Review by Keksoj — CHANGES_REQUESTED** [review]
i trust it works on your machine.

@fredalix would you have time to check that this branch builds on freebsd? I know you use it :-)

```
sudo pkg install rust
git clone git@github.com:yonasBSD/sozu.git
cd sozu
cargo build
```

Only if you have the time :pray:

**Pull request closed without merging**